checkov/arm/checks/resource/ACREnableZoneRedundancy.py

```diff
@@ -0,0 +1,37 @@
+from __future__ import annotations
+
+from checkov.common.models.enums import CheckCategories, CheckResult
+from checkov.arm.base_resource_check import BaseResourceCheck
+
+from typing import Any
+
+
+class ACREnableZoneRedundancy(BaseResourceCheck):
+
+    def __init__(self) -> None:
+        """
+        Zone redundancy provides resiliency and high availability to
+        a registry or replication resource in a specific region. Supported on Premium.
+        """
+        name = "Ensure Azure Container Registry (ACR) is zone redundant"
+        id = "CKV_AZURE_233"
+        supported_resources = ("Microsoft.ContainerRegistry/registries","Microsoft.ContainerRegistry/registries/replications")
+        categories = (CheckCategories.BACKUP_AND_RECOVERY,)
+        super().__init__(name=name, id=id, categories=categories, supported_resources=supported_resources)
+
+    def scan_resource_conf(self, conf: dict[str, list[Any]]) -> CheckResult:
+        # check registry. default=false
+        if conf.get("properties").get("zoneRedundancy", []) != [True]:
+            return CheckResult.FAILED
+
+        # check each replica. default=false
+        replications = conf.get("replications", {})
```

Comment:
I don't see the `replications` field in `Microsoft.ContainerRegistry/registries/replications` or `Microsoft.ContainerRegistry/registries`, are you sure this is the field we should check here?

Reply:
I changed the error
can you confirm me please

Reply:
My comment was for the resource template, Are you sure that those resources have the field `replications` to configure?

Reply:
I checked and I think it's good now

Reply:
You are looking for the field `replications` in the `Microsoft.ContainerRegistry/registries` resource type
but I don't see [here](https://learn.microsoft.com/en-us/azure/templates/microsoft.containerregistry/registries?pivots=deployment-language-arm-template) the `replications` filed